File: app/src/main/java/com/fullmeadalchemist/mustwatch/MustWatchApp.java
package com.fullmeadalchemist.mustwatch;

import android.app.Activity;
import android.app.Application;

import com.fullmeadalchemist.mustwatch.di.AppInjector;

import javax.inject.Inject;

import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;


public class MustWatchApp extends Application implements HasActivityInjector {

    public static final String MUST_WATCH_SHARED_PREFS = "MUST_WATCH_SHARED_PREFS";

    @Inject
    DispatchingAndroidInjector<Activity> dispatchingAndroidInjector;

    @Override
    public void onCreate() {
        super.onCreate();
//        if (BuildConfig.DEBUG) {
//            Timber.plant(new Timber.DebugTree());
//        }
        AppInjector.init(this);
    }

    @Override
    public DispatchingAndroidInjector<Activity> activityInjector() {
        return dispatchingAndroidInjector;
    }
}

File: app/src/main/java/com/fullmeadalchemist/mustwatch/MustWatchApp.kt
/*
 * Copyright (c) 2017 Full Mead Alchemist, LLC.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.fullmeadalchemist.mustwatch

import android.support.multidex.MultiDexApplication
import com.crashlytics.android.Crashlytics
import com.crashlytics.android.core.CrashlyticsCore
import com.facebook.stetho.Stetho
import com.fullmeadalchemist.mustwatch.BuildConfig.DEBUG
import com.fullmeadalchemist.mustwatch.core.appModule
import com.jakewharton.threetenabp.AndroidThreeTen
import io.fabric.sdk.android.Fabric
import org.koin.android.ext.android.startKoin
import timber.log.Timber


class MustWatchApp : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()

        if (DEBUG) {
            Stetho.initializeWithDefaults(this)
            Timber.plant(Timber.DebugTree())
        }
        initFabric()
        startKoin(this, listOf(appModule))
        AndroidThreeTen.init(this)
    }

    private fun initFabric() {
        val crashlyticsKit = Crashlytics.Builder()
                .core(CrashlyticsCore.Builder().disabled(BuildConfig.DEBUG).build())
                .build()

        // Initialize Fabric with the debug-disabled crashlytics.
        Fabric.with(this, crashlyticsKit)
    }

    companion object {
        const val MUST_WATCH_SHARED_PREFS = "MUST_WATCH_SHARED_PREFS"
    }
}
